src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+
+using System;
+using System.Net.Http;
+using System.Net;
+
+namespace System.Net.Http
+{
+    sealed internal class HttpEnvironmentProxy : IWebProxy {
+        private Uri _http = null;    // String URI for HTTP requests
+        private Uri _https = null;   // String URI for HTTPS requests
+        private NetworkCredential _httpCred;
+        private NetworkCredential _httpsCred;
+        String[] _bypass = null;    // list of domains not to proxy
+
+
+        public static HttpEnvironmentProxy TryToCreate()
+        {
+            string http = null;     // local variable to hold most specific value
+            string https = null;
+
+
+            // get environmental variables. protocol specific take precedence over
+            // general all_*, lover case variable has precedence over upper case.
+            // note that curl uses HTTPS_PROXY but not HTTP_PROXY.
+            // For http, only http_proxy and generic variables are used.
+            string value = Environment.GetEnvironmentVariable("ALL_PROXY");
+            if (value != null)
+            {
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("all_proxy");
+            if (value != null)
+            {
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("http_proxy");
+            if (value != null)
+            {
+                http = value;
+            }
+            value = Environment.GetEnvironmentVariable("HTTPS_PROXY");
+            if (value != null)
+            {
+                https = value;
+            }
+
+            value = Environment.GetEnvironmentVariable("https_proxy");
+            if (value != null)
+            {
+                https = value;
+            }
+            // fail to instantiate if nothing is set.
+            // caller may pick some other proxy type.
+            if (String.IsNullOrWhiteSpace(http) && String.IsNullOrWhiteSpace(https))
+            {
+                return null;
+            }
+
+            return new HttpEnvironmentProxy(http, https);
+        }
+
+        private HttpEnvironmentProxy(string http, string https)
+        {
+
+            if (!String.IsNullOrWhiteSpace(http))
+            {
+                _http = GetUriFromString(http);
+                if (!string.IsNullOrWhiteSpace(_http.UserInfo))
+                {
+                    String[] s = _http.UserInfo.Split(':', 2);
```

Comment:
Factor the code into a helper method to avoid duplicate code below? Also, consider using `IndexOf` and `Substring` instead of `Split` to avoid unnecessary allocations.